src/selection_edge.c

```diff
@@ -146,3 +125,44 @@ void selectionEdgeMotionDraw(int x0, int y0, int x1, int y1)
 
     selectionEdgeDraw();
 }
+
+void selectionEdgeDestroy(void)
+{
+    const struct Selection *const sel = *selectionGet();
+    struct SelectionEdge *pe = sel->edge;
+
+    if (pe->wndDraw != 0) {
+        waitUnmapWindowNotify();
+        XFree(pe->classHint);
+        XDestroyWindow(disp, pe->wndDraw);
+    }
+
+    free(pe);
+}
+
+static void waitUnmapWindowNotify(void)
+{
+    const struct Selection *const sel = *selectionGet();
+    const struct SelectionEdge *const pe = sel->edge;
+    XEvent ev;
+
+    XSelectInput(disp, pe->wndDraw, StructureNotifyMask);
+
+    XUnmapWindow(disp, pe->wndDraw);
+
+    struct timespec delay = {0, 80000000L}; // 80ms
+
+    for (short i = 0; i < 30; ++i) {
+        if (XCheckIfEvent(disp, &ev, &xeventUnmap, (XPointer) & (pe->wndDraw)))
+            break;
+        nanosleep(&delay, NULL);
+    }
+}
+
+
+static Bool xeventUnmap(Display *dpy, XEvent *ev, XPointer arg)
+{
+    (void)dpy; // unused
+    Window *win = arg;
```

Comment:
Generic pointers must be cast so the intention is well documented. I don't know why you changed it.
I'm going to change it before the merge.